### PR TITLE
cherry-pick #9528 MCP config: support tls mode simple/mutual/none

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
address https://github.com/istio/istio/pull/9482#pullrequestreview-168051663 part 2.

### CODE REVIEW

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -497,35 +497,58 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
...
-		if configSource.TlsSettings != nil && configSource.TlsSettings.Mode == istio_networking_v1alpha3.TLSSettings_ISTIO_MUTUAL {
-			requiredFiles := []string{
-				path.Join(model.AuthCertsPath, model.RootCertFilename),
-				path.Join(model.AuthCertsPath, model.CertChainFilename),
-				path.Join(model.AuthCertsPath, model.KeyFilename),
+		if configSource.TlsSettings != nil &&
+			configSource.TlsSettings.Mode != istio_networking_v1alpha3.TLSSettings_DISABLE {
+			var credentialOption *creds.Options
+			switch configSource.TlsSettings.Mode {
+			case istio_networking_v1alpha3.TLSSettings_SIMPLE:
```
Comment (ayj) [inline pilot/pkg/bootstrap/server.go]: Note that Galley doesn't support simple TLS.

Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/server.go]: We simply support this when users have their own server.

**Pull request merged**